### PR TITLE
Add GUI Integration into FSLeyes

### PR DESCRIPTION
**Pull request by ahill187**
## Checklist

- [x] I've given this PR a concise, self-descriptive, and meaningful title
- [x] I've linked relevant issues in the PR body
- [x] I've applied the relevant labels to this PR
- [ ] I've added relevant tests for my contribution
- [ ] I've updated the documentation and/or added correct docstrings
- [x] I've assigned a reviewer

## Description
We are integrating the CLI tools from `shimming-toolbox` into the `FSLeyes` package, as a plugin.

> ⚠️ When testing this PR, you should say "NO" when FSLeyes is asking to install this plugin permanently. Otherwise, you end us with this issue: https://github.com/shimming-toolbox/shimming-toolbox/pull/232#issuecomment-800697168

## Linked issues
https://github.com/shimming-toolbox/shimming-toolbox/issues/205

### CODE REVIEW

**Review by jcohenadad — COMMENTED** [review]
I do prefer scripts, but open to discussion -- see https://github.com/shimming-toolbox/shimming-toolbox/issues/205#issuecomment-790006679

**Comment by jcohenadad** [comment]
Few review comments (from high priority to low priority):
- Installing fsleyes and the plugin is cumbersome. Creating launchers instead of expecting the user to have the proper venv might be a better alternative: https://github.com/shimming-toolbox/shimming-toolbox/issues/237
- consider changing the plugin for a script: https://github.com/shimming-toolbox/shimming-toolbox/issues/205#issuecomment-790006679
- implement feature of selecting "input files" that are listed in the overlay section of fsleyes
- "mask" card: i got this error: `no such option: -i`
- "fieldmap": for unwrapper: create a combo box listing the options 
- "fieldmap": we should be able to only select the folder containing the fieldmaps-- the function will figure out each phase files and the number of echos
- "dcm2niix" card: if output folder does not exist, it will crash.
- add more verbose in the Terminal

File: shimmingtoolbox/utils.py [hunk 1]
```diff
@@ -100,3 +101,18 @@ def st_progress_bar(*args, **kwargs):
         kwargs['disable'] = True
 
     return tqdm.tqdm(*args, **kwargs)
+
+
+def create_output_dir(path_output, output_folder_name="output"):
```
Comment (po09i) [inline shimmingtoolbox/utils.py]:
```suggestion
def create_output_dir(path_output):
```

File: shimmingtoolbox/utils.py [hunk 2]
```diff
@@ -100,3 +101,18 @@ def st_progress_bar(*args, **kwargs):
...
+    """Given a path, create the directory if it doesn't exist.
+
+    Args:
+        path_output (str): Full path to either a folder or a file.
+        output_folder_name (str): Name of sub-folder.
```
Comment (po09i) [inline shimmingtoolbox/utils.py]:
```suggestion

```

**Comment by po09i** [comment]
`FSLEYES_PLUGIN_PATH` shell variable can be used to automatically load the newest version of the plugin. Only need to activate it to show it up.

**Comment by jcohenadad** [comment]
hum, why is it looking there? i'm using https://github.com/shimming-toolbox/shimming-toolbox/pull/232/commits/c5775938811eb1dfc308ad571649ab8bbdcd8d68

Screen Shot 2021-03-16 at 8 12 47 PM

**Comment by po09i** [comment]
> hum, why is it looking there? i'm using [c577593](https://github.com/shimming-toolbox/shimming-toolbox/commit/c5775938811eb1dfc308ad571649ab8bbdcd8d68)

Since we can't import any shimming toolbox packages, we also can't import `__dir_shimmingtoolbox__`. Two things rely on this: the value for default paths and the logo. 

We defined `__dir_shimmingtoolbox__` like so:
`__dir_shimmingtoolbox__ = os.path.dirname(os.path.dirname(os.path.dirname(os.path.realpath(__file__))))` which works if the gui file is in the shimming toolbox repo. I'm guessing it tried to use the GUI file that was in .fsleyes so it could not find the logo since it was the wrong shimming toolbox path.

The only solution I see is having a shell variable for Shimming Toolbox so that we can find the logo. However, I don't think this is relevant to fix since we want to move to a fsleyes package in #242 anyways and the package will include the logo and change the default paths to the current directory.

PS: To test, I would remove the files in the .fsleyes directory, relaunch fsleyes, load the plugin but select "no" when prompted to permanently install since we are still in the dev stage. Hopefully it should work after that.

**Comment by po09i** [comment]
It looks good to me but I was a big contributor so I'll wait for others to have a look.

**Comment by jcohenadad** [comment]
> > hum, why is it looking there? i'm using [c577593](https://github.com/shimming-toolbox/shimming-toolbox/commit/c5775938811eb1dfc308ad571649ab8bbdcd8d68)
> 
> Since we can't import any shimming toolbox packages, we also can't import `__dir_shimmingtoolbox__`. Two things rely on this: the value for default paths and the logo.
> 
> We defined `__dir_shimmingtoolbox__` like so:
> `__dir_shimmingtoolbox__ = os.path.dirname(os.path.dirname(os.path.dirname(os.path.realpath(__file__))))` which works if the gui file is in the shimming toolbox repo. I'm guessing it tried to use the GUI file that was in .fsleyes so it could not find the logo since it was the wrong shimming toolbox path.
> 
> The only solution I see is having a shell variable for Shimming Toolbox so that we can find the logo. However, I don't think this is relevant to fix since we want to move to a fsleyes package in #242 anyways and the package will include the logo and change the default paths to the current directory.
> 
> PS: To test, I would remove the files in the .fsleyes directory, relaunch fsleyes, load the plugin but select "no" when prompted to permanently install since we are still in the dev stage. Hopefully it should work after that.

what i did:
- removed the file `~/.fsleyes/plugins/st_plugin.py`
- activated ST venv
- opened fsleyes
- re-loaded this plugin
- i get the same error

**Review by gaspardcereza — APPROVED** [review]
We tested everything from scratch and it worked (MacOs Catalina & Big Sur). Installation process will be updated once the new plugin repo will be working.

**Review by po09i — CHANGES_REQUESTED** [review]
Before merging in master, we should merge #242 in this branch and port to the new repo the gui files, then remove those files from this repo.

**Pull request closed without merging**